src/components/Layouts/Shared/getNavPages.tsx

```diff
@@ -0,0 +1,223 @@
+import { LinkProps } from 'next/link';
+import { ReactElement, useMemo } from 'react';
+import CompassIcon from '@mui/icons-material/Explore';
+import { useTranslation } from 'react-i18next';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { reportNavItems } from './reportNavItems';
+import { settingsNavItems } from './settingsNavItems';
+import { toolsNavItems } from './toolsNavItems';
+
+interface SubItems {
+  id?: string;
+  title?: string;
+  grantedAccess?: string[];
+}
+interface Item {
+  id?: string;
+  href?: LinkProps['href'];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+  items?: Item[];
+  title: string;
+  subtitle?: string;
+  desc?: string;
+  // search dialog specific
+  showInSearchDialog?: boolean;
+  searchIcon?: ReactElement;
+  searchName?: string;
+  // settings specific
+  grantedAccess?: string[];
+  subItems?: SubItems[];
+  oauth?: boolean;
+}
+
+export interface NavPage {
+  id?: string;
+  title: string;
+  href?: LinkProps['href'];
+  pathname?: string;
+  searchIcon?: ReactElement;
+  items?: Item[];
+  whatsNewLink?: boolean;
+  showInNav?: boolean;
+  showInSearchDialog?: boolean;
+  showInPanel?: boolean;
+}
+
+export function getNavPages(
```

Comment:
This function uses React hooks, so that makes it a hook itself.

```suggestion
export function useNavPages(
```